src/components/DateRangeInput/DateRangeInput.tsx

```diff
@@ -126,9 +126,16 @@ const DateRangeInput: React.FC<
 
   const onApply = useCallback(
     e => {
+      // To avoid inconsistent results round the start and end dates to the
+      // nearest minute (or day if the time picker is disabled)
+      const timeUnit = withTime ? 'minute' : 'day';
+      const startEndDates = [
+        dayjs(currentDateRange[0]).startOf(timeUnit).toDate(),
+        dayjs(currentDateRange[1]).endOf(timeUnit).toDate(),
```

Comment:
I am not pretty sure if the `currentDateRange array has always two items, can we confirm this one?

Reply:
The Apply button is disabled until the users fill in both dates, is this enough or should I add an extra validation in the `onApply` function?

```
 const isDisabled = useCallback(() => !currentDateRange || currentDateRange?.length < 2, [
    currentDateRange,
  ]);
```
               
```
 <Button disabled={isDisabled()} onClick={onApply} size="medium">
       Apply
</Button>
```